src/kiara_plugin/topic_modelling/modules/corpus_preparation.py

```diff
@@ -109,4 +110,71 @@ def get_date(file):
         except Exception as e:
             print(f"An error occurred: {e}")
 
-        outputs.set_value("corpus_table", output_table)
\ No newline at end of file
+        outputs.set_value("corpus_table", output_table)
+
+class CorpusDistTime(KiaraModule):
+    """
+    This module aggregates a table by day, month or year from a corpus table that contains a date column. It returns the distribution over time, which can be used for display purposes, such as visualization.
+
+    Dependencies:
```

Comment:
This would probably be better added to the module metadata, like:

```python
class CorpusDistTime(KiaraModule):

    KIARA_METADATA = {
        "references": {
            "pyarrow": {
                "url": "https://arrow.apache.org/docs/python/"
            }
        }
    }
```

But to be honest, I don't think this needs to be in the module metadata at all, IMHO it's enough to add it to the source code comments if necessary. End-users would not know what to do with those links, or why they are given in the first place?

Reply:
In my understanding, this was emphasized as good practice by @stakats: listing the dependencies in the module documentation. I guess that it is for transparency reasons. I will add this point as an agenda item at the next dev meeting.

Reply:
But this would be a very random selection of dependencies and I don't really see the point. The relevant place to document the arrow dependency would be the data-type, if any, but even there. Each value that is created within kiara has a list of all Python packages within the virtual environment, so that would be a much better place to look, right? I don't have a strong opinion whether to add this, but it feels a lot like pretending to be transparent, because there is not much one can do with that information. Either you read the module code and understand what is happening in more detail, or you trust that it does what the doc says it does, but what do you do with the information that at some point in the code some dependency is used? We'd have to add a lot of those to each module...

Reply:
I also do not really see the point, and I do not have a strong opinion about this, but I will ask the question at next dev meeting.